data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java

```diff
@@ -138,6 +144,21 @@ public static ConnectionStringBuilder createWithUserPrompt(String resourceUri, S
         return csb;
     }
 
+    public static ConnectionStringBuilder createWithDeviceAuthentication(String resourceUri) {
```

Comment:
createWithDeviceCode or createWithDeviceCodeAuthentication (also next method)

Though most of the Java SDK methods don't include the word "Authentication", the .NET and Python SDKs do. So the latter probably makes more sense for the sake of consistency.